modules/development/nix.nix

```
@@ -13,10 +15,22 @@ in
     };
 
     config = mkIf cfg.enable {
-      nix.settings.experimental-features = ["nix-command" "flakes"];
-      nix.extraOptions = ''
-        keep-outputs          = true
-        keep-derivations      = true
-      '';
+      nix = {
+        settings.experimental-features = ["nix-command" "flakes"];
+        extraOptions = ''
+                       keep-outputs          = true
+          keep-derivations      = true
+        '';
+        # Set the path and registry so that e.g. nix-shell and repl work
+        nixPath = [
+          "nixpkgs=${pkgs.path}"
```

Avoid `pkgs.path` here because it will create another expensive copy of nixpkgs at eval time:

```suggestion
          "nixpkgs=${inputs.nixpkgs}"
```